src/koa-node-resolve.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import * as Koa from 'koa';
+import {resolve as resolvePath} from 'path';
+import {parse as parseURL} from 'url';
+
+import esmSpecifierTransform from './koa-esm-specifier-transform';
+import {getBasePath, noLeadingSlash} from './support/path-utils';
+import resolveSpecifier from './support/resolve-node-specifier';
+
+export type Options = {
+  /* On-disk package root path used for NPM package resolution; defaults to
+     current folder. */
+  packageRoot?: string,
+  /* When serving the package from a base other than root.  If this option is
+     given, no specifier resolution will be attempted for documents served
+     outside the baseHref. */
+  baseHref?: string,
+};
+
+export const middleware = (options: Options = {}): Koa.Middleware => {
+  const onDiskPackageRoot = resolvePath(options.packageRoot || '.');
+  const basePath =
+      options.baseHref ? noLeadingSlash(getBasePath(options.baseHref)) : '';
+  return esmSpecifierTransform((baseURL: string, specifier: string) => {
+    const pathname = parseURL(baseURL).pathname || '/';
+    const path = noLeadingSlash(pathname);
+    if (!path.startsWith(basePath)) {
```

Comment:
I tried using koa-mount in the karma configuration but it was kind of puzzling to set up right in a   proxy scenario.

I'll tell you what, I'll have another go at it to see if I was just generally confused the day I was trying it.

Reply:
So, the problem is that `koa-mount` is used to mount an entire Koa application, which means the only way to leverage it is to construct multiple proxy servers too:

```js
const server = new Koa()
  .use(mount('/base', new Koa()
        .use(middleware())
        .use(proxy({ host: 'http://127.0.0.1:9876', map: p => 'base/' + p }))))
  .use(proxy({ host: 'http://127.0.0.1:9876' }))
  .listen(9877)
```

And even this has issues, like I have to re-prefix 'base' when I get to the proxy after the middleware because the karma server is the one serving files found in `/base/`.

In the end, I think a `baseHref` option in the middleware is a reasonable amount of flexibility, especially since the `packageRoot` option might also specify something other than the web root and these two options probably will wind up supporting each other in different configurations.

Reply:
Oh wut-- I just tried the following and it worked

```js
const server = new Koa()
  .use(mount('/base', new Koa().use(middleware())))
  .use(proxy({ host: 'http://127.0.0.1:9876' }))
  .listen(9877)
```

🤦🏼‍♂️

So I guess I didn't realize Koa applications themselves can be used as middleware and `next()` will just work.  *huh*

Reply:
I'm going to nix the `baseHref` option.

Reply:
Okay in other news, I didn't even try just mounting the middleware itself and that totally worked so I guess I'm not a Koa expert yet :-p
```js
const server = new Koa()
  .use(mount('/base', middleware()))
  .use(proxy({ host: 'http://127.0.0.1:9876' }))
  .listen(9877)
```

Reply:
Also yay thanks for suggesting koa-mount